include/git2/diff.h

```diff
@@ -381,6 +399,8 @@ typedef struct {
 	git_strarray       pathspec;     /**< defaults to include all paths */
 	git_diff_notify_cb notify_cb;
 	void              *notify_payload;
+	git_diff_progress_cb progress_cb;
+	void                *progress_payload;
```

Comment:
I don't think we should have different payloads for each callback. I'd much rather see `notify_payload` be renamed to `payload`.